### ISSUE TITLE
Allow general headers in HTTP requests

### ISSUE THREAD

**Issue by mcsimps2** [Digital-Sapphire/PyUpdater#219]
**Problem:** Currently, the only headers we allow users to specify, through their `client_config.py`, are those that can be passed to `urllib3.util.make_headers`, namely `keep_alive`, `accept_encoding`, `user_agent`, and `basic_auth`.  For more advanced usage, it is incredibly useful to enable the ability to pass headers besides these 4.  For example, this might include other types of the `Authorization` header besides basic authentication (e.g. `Bearer`).

**Proposed Solution:** In `downloader.py`, pass these other headers in the `_http.headers.update()` method.
```
def _get_http_pool(self, secure=True):
        if secure:
            _http = urllib3.PoolManager(
                cert_reqs=str("CERT_REQUIRED"), ca_certs=certifi.where()
            )
        else:
            _http = urllib3.PoolManager()

        if self.headers:
            _headers = urllib3.util.make_headers(**self.headers)
            _http.headers.update(_headers)
        log.debug(_http.headers)
        return _http
```
To avoid breaking changes, perhaps this is best done by partitioning `self.headers` into two dictionaries, one to record the values of the urllib3.utils.make_headers compatible keys (`keep_alive`, `accept_encoding`, `user_agent`, and `basic_auth`) and pass those to `make_headers`, and the rest can be passed as normal.

Excuse the Python:
```
def _get_http_pool(self, secure=True):
        if secure:
            _http = urllib3.PoolManager(
                cert_reqs=str("CERT_REQUIRED"), ca_certs=certifi.where()
            )
        else:
            _http = urllib3.PoolManager()

        if self.headers:
              urllib_header_keys = inspect.getargspec(func)[2] # Need to import inspect, but this will return keep_alive, accept_encoding, user_agent, basic_auth
              urllib_header = {}
              other_headers = {}
              for k, v in self.headers.items():
                     if k in urllib_header_keys:
                              urllib_headers[k] = v
                     else:
                              other_headers[k] = v
              _http.headers.update(urllib3.util.make_headers(**urllib_headers))
              _http.headers.update(other_headers)
        log.debug(_http.headers)
        return _http
```

The hope is that users can do something like this:
```
client = Client(client_config,
                    refresh=True,
                    progress_hooks=[print_status_info],
                    headers={
                        "keep_alive": True,  # Parsed by urllib3 make_headers
                        "Authorization": "Bearer <auth_token>"  # Not parsed by urllib3 make_headers
                    })
```

**Comment by JMSwag** [comment]
@mcsimps2 Another one! #DJKhaledVoice Great suggestion. 

I’d happily accept a PR.

**Comment by mcsimps2** [comment]
Currently working on a PR as of last week, will push soon!

**Comment by mcsimps2** [comment]
Closed by #226
